h2o-algos/src/test/java/hex/tree/TreeHandlerTest.java

```diff
@@ -0,0 +1,221 @@
+        GBMModel model = null;
+
+        Scope.enter();
+        try {
+            tfr = parse_test_file("./smalldata/iris/iris2.csv");
+            DKV.put(tfr);
+            GBMModel.GBMParameters parms = new GBMModel.GBMParameters();
+            parms._train = tfr._key;
+            parms._response_column = "response";
+            parms._ntrees = 1;
+            parms._seed = 0;
+
+            model = new GBM(parms).trainModel().get();
+            final SharedTreeModel.SharedTreeOutput sharedTreeOutput = model._output;
+            assertEquals(parms._ntrees, sharedTreeOutput._ntrees);
+            assertEquals(parms._ntrees, sharedTreeOutput._treeKeys.length);
+            assertEquals(parms._ntrees, sharedTreeOutput._treeKeysAux.length);
+
+            final int treeNumber = 0;
+            final int treeClass = 0;
+            final CompressedTree auxCompressedTree = sharedTreeOutput._treeKeysAux[treeNumber][treeClass].get();
+            final SharedTreeSubgraph sharedTreeSubgraph = sharedTreeOutput._treeKeys[treeNumber][treeClass].get()
+                    .toSharedTreeSubgraph(auxCompressedTree, sharedTreeOutput._names, sharedTreeOutput._domains);
+
+            assertNotNull(sharedTreeSubgraph);
+
+            final TreeHandler.TreeProperties treeProperties = TreeHandler.convertSharedTreeSubgraph(sharedTreeSubgraph);
+            assertNotNull(treeProperties);
+
+            final String[] nodeDescriptions = treeProperties._descriptions;
+            assertEquals(sharedTreeSubgraph.nodesArray.size(), nodeDescriptions.length);
+
+            for (String nodeDescription : nodeDescriptions) {
+                assertFalse(nodeDescription.isEmpty());
+            }
+
+        } finally {
+            if (tfr != null) tfr.remove();
+            if (model != null) model.remove();
+        }
+
+    }
+
+    @Test
+    public void testSharedTreeSubgraphConversion_argumentValidation() {
+
+        Frame tfr = null;
+        GBMModel model = null;
+
+        Scope.enter();
+        try {
+            tfr = parse_test_file("./smalldata/iris/iris2.csv");
+            DKV.put(tfr);
+            GBMModel.GBMParameters parms = new GBMModel.GBMParameters();
+            parms._train = tfr._key;
+            parms._response_column = "response";
+            parms._ntrees = 1;
+            parms._seed = 0;
+
+            model = new GBM(parms).trainModel().get();
+
+            final TreeHandler treeHandler = new TreeHandler();
+            TreeV3 args = new TreeV3();
+            args.model = new KeyV3.ModelKeyV3(Key.make());
+
+            //Unexisting key
+
+            boolean exceptionThrown = false;
+            try {
+                treeHandler.getTree(3, args);
+            } catch (IllegalArgumentException e) {
+                assertTrue(e.getMessage().contains("Given model does not exist"));
+                exceptionThrown = true;
+            }
+            assertTrue(exceptionThrown);
+            exceptionThrown = false;
+
```

Comment:
I tend to use a simpler pattern for this that doesn't need to maintain a variable:
```
try {
   do_stuff_that_should_throw_exception();
   fail("should have thrown exception");
} catch (Exception expected) {
    //assert expected message here
}
```

Reply:
Indeed better. Acknowledged.